arch/simddetect.cpp

```diff
@@ -26,7 +26,7 @@
 #endif // x86 target
 
 #if defined(X86_BUILD)
-# if defined(__linux__) || defined(__MINGW32__) || defined(__APPLE__)
+# if defined(__GNUC__) || defined(__clang__) || defined(__MINGW32__) || defined(__MINGW64__) 
```

Comment:
MinGW-w64 64  defines `__MINGW32__` so ` || defined(__MINGW64__) ` is not necessary.

` (__GNUC__)` is defined not only by GCC, but also by Clang and ICC.